src/foraging_gui/RigJsonBuilder.py

```diff
@@ -0,0 +1,510 @@
+            if len(differences['values_changed']) == 0:
+                differences.pop('values_changed')
+
+    # Determine which schema to use
+    if len(differences) > 0:
+        # If any differences remain, rename the temp file
+        logging.info('differences with existing rig json: {}'.format(differences))
+        # Write to file 
+        time_str = datetime.now().strftime('%Y-%m-%d_%H_%M_%S')
+        filename = 'rig_{}_{}.json'.format(settings['rig_name'],time_str )
+        final_path = os.path.join(settings['rig_metadata_folder'],filename)
+        os.rename(new_rig_json_path, final_path)
+        logging.info('Saving new rig json: {}'.format(filename))
+    else:
+        # Delete temp file
+        os.remove(new_rig_json_path)
+        logging.info('Using existing rig json')
+
+
+def build_rig_json_core(settings, water_calibration, laser_calibration):
+
+    # Set up
+    ###########################################################################
+    logging.info('building rig json')
+
+    # Build dictionary of components
+    components = {}
+
+    # TODO, what other modalities do we need to include?
+    # behavior video?
+    FIB = settings['Teensy_COM_box{}'.format(settings['box_number'])] != ''
+    OPTO = False
+
+    # Modalities
+    ###########################################################################
+    components['modalities'] = [Modality.BEHAVIOR]
+    if FIB:
+        components['modalities'].append(Modality.FIB)
+
+    # Cameras
+    ###########################################################################
+    # TODO, ensure this toggles with rig
+    components['cameras']=[
+        d.CameraAssembly(
+            name="BehaviorVideography_FaceSide",
+            #camera_assembly_name="BehaviorVideography_FaceBottom",
+            camera_target=d.CameraTarget.FACE_SIDE_RIGHT,
+            camera=d.Camera(
+                name="Side face camera",
+                detector_type="Camera",
+                serial_number="TBD",
+                manufacturer=d.Organization.AILIPU,
+                model="ELP-USBFHD05MT-KL170IR",
+                notes="The light intensity sensor was removed; IR illumination is constantly on",
+                data_interface="USB",
+                computer_name="W10DTJK7N0M3",
+                max_frame_rate=120,
+                sensor_width=640,
+                sensor_height=480,
+                chroma="Color",
+                cooling="Air",
+                bin_mode="Additive",
+                recording_software=d.Software(name="Bonsai", version="2.5"),
+            ),
+            lens=d.Lens(
+                name="Xenocam 1",
+                model="XC0922LENS",
+                serial_number="unknown",
+                manufacturer=d.Organization.OTHER,
+                max_aperture="f/1.4",
+                notes='Focal Length 9-22mm 1/3" IR F1.4',
+            ),
+        ),
+        d.CameraAssembly(
```

Comment:
The camera assembly could be varied (e.g. high-speed camera assembly and the current camera assembly used in 447). For high-speed camera assembly, the side and bottom cameras used different lenses and filters.

Reply:
This is resolved now